File: 15-trees1/1-intro/main.py
# Tree Node Definition - Python Example

class TreeNode:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None

# Creating a simple binary tree
root = TreeNode(1)
root.left = TreeNode(2)
root.right = TreeNode(3)

print("Root node:", root.data)
print("Left child:", root.left.data)
print("Right child:", root.right.data)

File: 15-trees1/1-intro/main.cpp
// Tree Node Definition - C++ Example

#include <iostream>

struct TreeNode {
    int data;
    TreeNode* left;
    TreeNode* right;
    TreeNode(int val) : data(val), left(nullptr), right(nullptr) {}
};

int main() {
    TreeNode* root = new TreeNode(1);
    root->left = new TreeNode(2);
    root->right = new TreeNode(3);

    std::cout << "Root node: " << root->data << std::endl;
    std::cout << "Left child: " << root->left->data << std::endl;
    std::cout << "Right child: " << root->right->data << std::endl;

    return 0;
}
